httphandler.go

```diff
@@ -44,6 +46,67 @@ func (s *MessagingServer) StartServer() error {
 	return nil
 }
 
+func (s *MessagingServer) handleLogIssue(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	description := strings.TrimSpace(r.FormValue("description"))
+	clientName := strings.TrimSpace(r.FormValue("name"))
+	clientEmail := strings.TrimSpace(r.FormValue("email"))
+	clientPhoneNumber := strings.TrimSpace(r.FormValue("phone"))
+	callback := strings.TrimSpace(r.FormValue("callback"))
+	browserName := strings.TrimSpace(r.FormValue("browser"))
+	browserVersion := strings.TrimSpace(r.FormValue("browserVersion"))
+	serverAddress := strings.TrimSpace(r.FormValue("serverAddress"))
+
+	summary := fmt.Sprintf("Issue detected at %v", serverAddress)
+
+	messageTemplate := `
+	Name: %v
+	Email: %v
+	Phone number:%v
+	Call me back: %v
+	Browser: %v
+	Browser version: %v
+	Server: %v
+	
+	Message:					
+	%v
+	`
+	message := fmt.Sprintf(messageTemplate, clientName, clientEmail, clientPhoneNumber, callback, browserName, browserVersion, serverAddress, description)
+
+	issueKey, err := s.JiraApi.CreateIssue(summary, message)
+	if err != nil {
+		s.Log.Infof("Unable to log issue with the Jira Api: %v", err.Error())
+		http.Error(w, "Unable to log issue with the Jira Api", http.StatusInternalServerError)
+		return
+	}
+
+	ct := r.Header.Get("Content-Type")
+	contentTypeList := strings.Split(ct, ";")
+	contentType := contentTypeList[0]
+
+	var statusCode int
+	if contentType == "multipart/form-data" && issueKey != "" {
+		statusCode, err = s.JiraApi.AddAttachments(issueKey, r)
+		if err != nil {
+			s.Log.Infof("Unable to add attachments to Jira ticket with Id: %v, error: %v", issueKey, err.Error())
+			http.Error(w, fmt.Sprintf("Unable to add attachments to Jira ticket with Id: %v, error: %v", issueKey, err.Error()), statusCode)
+		}
+	}
+
+	type successJson struct {
```

Comment:
I think it's standard practice to define the types at the start of the module file (I could be wrong).